src/registrar/models/domain.py

```diff
@@ -1322,6 +1336,11 @@ def ready(self):
         """
         logger.info("Changing to ready state")
         logger.info("able to transition to ready state")
+        # if self.first_ready_at is not None, this means that his
+        # domain wasr READY, then not READY, then is READY again.
+        # We do not want to overwrite first_ready_at.
+        if self.first_ready_at is None:
+            self.first_ready_at = timezone.now()
```

Comment:
How would first_ready be set to None?

Reply:
Also, I just noticed all the typos in my comment. Will fix.

Reply:
@rachidatecs The interaction would have to be external, like for instance a script or maybe an admin makes an accidental edit. Its not worth adding a fix for it like I mentioned because the cause would be unrelated to this code. 

Comment wise, it'd just be helpful if some problematic code gets added down the line and we see some weird behavior on this property. Might save some time just to note that it _could_ happen.

That said, this suggestion is more of a general comment/nice to have so its not a big deal either way

Reply:
Totally worth discussing. `first_ready_at` is `editable=False`, so an admin or a dev would really want to break things.